app/services/importers/ok_assessor/sales.rb

```diff
@@ -0,0 +1,47 @@
+require 'csv'
+
+module Importers
+  module OkAssessor
+    class Sales < BaseImporter
+      attr_reader :accounts
+
+      def attributes
+        {
+          account_id: accounts[row['ACCOUNT_NUM']],
+          reception_number: row['RECEPTION_NUM'],
+          grantor: row['GRANTOR'],
+          grantee: row['GRANTEE'],
+          sale_price: row['SALE_PRICE'],
+          sale_date: parse_date(row['SALE_DATE']),
+          deed_type: row['DEED_TYPE'],
+          valid_sale: row['VALID'],
+          confirm: row['CONFIRM'],
+          book: row['BOOK'],
+          page: row['PAGE'],
+          revenue_stamps: row['REVENUE_STAMPS'],
+          change_date: parse_date(row['CHANGE_DATE'])
+        }
+      end
+
+      def validate_attributes!
+        raise AttributeError, "No account found for number: #{row['ACCOUNT_NUM']}" if attributes[:account_id].nil?
+      end
+
+      def unique_by
+        [:account_id, :reception_number]
+      end
```

Comment:
In general there are a lot of unique_by constraints that still result with duplicates in batches. I confirmed that most of them are truly duplicates. This was probably the only one I considered tweaking.

My guess is that this key is largely correct but if a sale is attempted between two parties and fails i think a new one (between two different parties) might come through with the same reception number. I was seeing this for about 1/100 sales and currently they're being ignored as duplicates.